code/modules/mob/living/simple_animal/hostile/mining_mobs.dm

```diff
@@ -812,6 +812,14 @@
 	visible_message("<span class='warning'>[name] burrows into the flesh of [H]!</span>")
 	var/mob/living/simple_animal/hostile/asteroid/hivelord/legion/L = new(H.loc)
 	visible_message("<span class='warning'>[L] staggers to their feet!</span>")
+	if(isdwarf(H))
+		L.name = "dwarf legion"
+		L.desc = "You can still see what was once a dwarf under the shifting mass of corruption."
```

Comment:
Regular legion is tall, dwarves are not. Having a humanoid-sized legion die and out pop a dwarf is kinda weird looking.